FloripaBus/ViewModels/DetailsViewModel.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.ObjectModel;
+
+namespace FloripaBus
+{
+	public class DetailsViewModel : ViewModelBase
+	{
+		private readonly IRouteRepository _routeRepository;
+
+		private Route _route;
+		public Route Route {
+			get {
+				return _route;
+			}
+			set {
+				_route = value;
+				Notify (() => Route);
+			}
+		}
+
+		public string Title {
+			get { return String.Format ("{0} - {1}", Route.ShortName, Route.LongName); }
+		}
+
+		private ObservableCollection<RouteStop> _routeStops;
+		public ObservableCollection<RouteStop> RouteStops {
+			get {
+				return _routeStops;
+			}
+			set {
+				_routeStops = value;
+				Notify (() => RouteStops);
+			}
+		}
+
+		private ObservableCollection<RouteDeparture> _weekDayDepartures;
+		public ObservableCollection<RouteDeparture> WeekDayDepartures {
+			get {
+				return _weekDayDepartures;
+			}
+			set {
+				_weekDayDepartures = value;
+				Notify (() => WeekDayDepartures);
+			}
+		}
+
+		private ObservableCollection<RouteDeparture> _saturdayDepartures;
+		public ObservableCollection<RouteDeparture> SaturdayDepartures {
+			get {
+				return _saturdayDepartures;
+			}
+			set {
+				_saturdayDepartures = value;
+				Notify (() => SaturdayDepartures);
+			}
+		}
+
+		private ObservableCollection<RouteDeparture> _sundayDepartures;
+		public ObservableCollection<RouteDeparture> SundayDepartures {
+			get {
+				return _sundayDepartures;
+			}
+			set {
+				_sundayDepartures = value;
+				Notify (() => SundayDepartures);
+			}
+		}
+
+		private bool _isLoading;
+		public bool IsLoading {
+			get {
+				return _isLoading;
+			}
+			set {
+				_isLoading = value;
+				this.ShowData = !_isLoading;
+				Notify (() => IsLoading);
+			}
+		}
+
+		private bool _showData;
+		public bool ShowData {
+			get {
+				return _showData;
+			}
+			set {
+				_showData = value;
+				Notify (() => ShowData);
+			}
+		}
+
+		public DetailsViewModel (Route route, IRouteRepository routeRepository)
+		{
+			_routeRepository = routeRepository;
+
+			this.Route = route;
+			this.LoadAsync ();
+		}
+
+		public async void LoadAsync()

```

Comment:
In most cases we should avoid using async methods that return `void`. Here is a good resource explaining why: http://haacked.com/archive/2014/11/11/async-void-methods/

Also, it is not good to call methods that i.e. load data in the constructor. In mobile a good place to call it is in the `NavigatedTo`, `ViewDidLoad`, etc. (depending on the platform) events, for example.